### DISCUSSION TITLE
RDF Files and Updates?

### DISCUSSION THREAD

**Discussion by megamorphg** [xiaoxiaoflood/firefox-scripts#187] [General Support]
I installed the XPIs for the addons as shown in the video and everything is working fine.

I am wondering what the RDF files are for... are those just for the older versions of FF?

Also wondering, how will we know when there is an update?
Sounds like we are going to need to manually download the files to update every time for both the XPIs and JSes and just watch this repository.

**Comment by megamorphg** [comment]
Also, I'm curious how we would uninstall the scripts complete? Just in case if I use it in my main browser (non-developer edition) and there is some issue.
Do we just remove the files that were added to the program directory and the files in the chrome directory?
Are there any other configs/files created that need to be addressed for a clean uninstall?

**Comment by xiaoxiaoflood** [comment]
RDF contains extension metadata like name, version, author, compatibility and update URL. They are like WebExtensions' `manifest.json`.

Firefox fetches for legacy extensions updates just like any other extension.

There's no auto update for userChromeJS, it's up to the user to follow this repository.

Legacy extensions don't remove leftovers on uninstall, whose range can go from `about:config` entries to files created in any directory Firefox has writing permission. There's no simple way to know which files were created by a legacy extension.

**Reply by megamorphg** [reply]
Ah so the legacy extensions will be able to auto-update but the scripts won't.

> There's no simple way to know which files were created by a legacy extension.

That makes me a little more careful to install any legacy extensions to my main profile.
I'm mainly interested in the FastTabSwitcher and SaveFIleTo extensions--not anything outside of your suggestions here.

Is this risk of legacy extensions the same for your scripts? Is there any possibility of it corrupting anything and/or what files does it create outside of the /chrome directory? Does it store its data in just browser-extension-data and cache?

I'd really like to install the scripts at least onto my main profile (non-developer) and safely play with toggling things on-off.
